client/allocrunner/taskrunner/task_runner.go

```diff
@@ -1335,10 +1335,14 @@ func (tr *TaskRunner) emitStats(ru *cstructs.TaskResourceUsage) {
 
 	if ru.ResourceUsage.MemoryStats != nil {
 		tr.setGaugeForMemory(ru)
+	} else {
+		tr.logger.Debug("Skipping memory stats for allocation", "reason", "MemoryStats is nil")
```

Comment:
I would adjust the casing, we typically have lower case message.  Also, `MemoryStats is nil` is low level implementation; would be nice to have a user facing message, maybe something like:
```suggestion
		tr.logger.Debug("memory stats for alloc is unpopulated; skipping publishing them")
```